console/network/environment/src/helpers/sanitizer.rs

```diff
@@ -117,12 +117,33 @@ impl Sanitizer {
     ///
     /// Discard any leading newline.
     fn str_till_eol(string: &str) -> ParserResult<&str> {
-        map(
-            recognize(Self::till(alt((value((), tag("\\\n")), value((), Sanitizer::parse_safe_char))), Self::eol)),
-            |i| {
-                if i.as_bytes().last() == Some(&b'\n') { &i[0..i.len() - 1] } else { i }
-            },
-        )(string)
+        // A heuristic approach is applied here in order to avoid
+        // costly parsing operations in the most common scenarios.
+        if let Some((before, after)) = string.split_once('\n') {
+            let is_multiline = before.ends_with('\\');
+
+            if !is_multiline {
+                let contains_unsafe_chars = !before.chars().all(is_char_supported);
+
+                if !contains_unsafe_chars {
+                    Ok((after, before))
```

Comment:
Is the goal to avoid allocations in [this](https://github.com/rust-bakery/nom/blob/54557471141b73ca3b2d07be88d6709a43495b10/src/multi/mod.rs#L287) method?

Reply:
IIRC the allocations that are avoided are primarily:
- the ones from [`verify`](https://docs.rs/nom/latest/src/nom/combinator/mod.rs.html#428)
- the ones from [`recognize`](https://docs.rs/nom/latest/src/nom/combinator/mod.rs.html#513)
- the ones from [`value`](https://docs.rs/nom/latest/src/nom/combinator/mod.rs.html#460)
- but most importantly, the `alt`-related ones from [here](https://github.com/AleoNet/snarkVM/pull/2563/files#diff-218cced7e98aba78cca56fdda8f839eb6a20a2414541d2805ec337d4e7e51d01R136) and [here](https://github.com/ProvableHQ/snarkVM/blob/3d42aa04a058cd5f46a1880b421313e1c04a63dc/console/network/environment/src/helpers/sanitizer.rs#L92)